src/core/components/LoginPage/index.js

```diff
@@ -20,7 +21,7 @@ export default class LoginPage extends React.Component {
           {message || _('You must be logged in to access this page.')}
         </p>
         <p>
-          <a className="button" href={startLoginUrl()}>
+          <a className="button" href={startLoginUrl({ location })}>
```

Comment:
Oh, I think I was mistaken. I was worried that this component won't handle the case where the user is already logged in but in tracing the use of this I see that it shouldn't be shown to already logged in users.